api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java

```diff
@@ -13,6 +13,48 @@
 
   List<Criteria> findCriteriaByType(@Param("type") String type);
 
+  @Query(value = "select * from criteria where id in ( " +
+    "select id from " +
+    "(select @curRow \\:= case " +
+    "when @curType = name " +
+    "then @curRow + 1 " +
+    "else 1 end as rank, " +
+    "id, " +
+    "code, " +
+    "@curType \\:= name as name, " +
+    "concept_id " +
+    "from (select * from criteria where type = upper(:type) and " +
+    "(upper(code) like upper(concat('%',:value,'%')) or upper(name) like upper(concat('%',:value,'%'))) ) a, " +
```

Comment:
We take a hit for upper casing functions, don't we? Is this really needed?

Reply:
Unfortunately, we need them

Reply:
ok, for protection against unknown data.